### PR TITLE
Guestbook changes. 

### PR DESCRIPTION
**Pull request by amygdala**
Use GCR images from 'google-samples' project, use
'worker' (not 'slave'), allow switch on whether dns service is
supported, or to use env vars to get service host info.

### CODE REVIEW

**Comment by amygdala** [comment]
Brendan, assigning initially to you as per our conversation.  See what you think about how I handled using dns service vs envir vars.  I ended up adding an env var in the 2 .yaml files that the image scripts use to determine which to use, with default being dns. So the docker images themselves work with both.

**Comment by amygdala** [comment]
This is towards #9107 and #9593.

**Comment by amygdala** [comment]
btw - I wouldn't think any of my changes to this example would have impacted the integration tests, since the previous version of the example was not causing any issues. Might need help figuring out whether the issues are spurious.

**Comment by brendandburns** [comment]
Errors are:
```console
 PATH=$GOPATH/bin:./third_party/etcd:$PATH ./hack/verify-gendocs.sh
docs/getting-started-guides/coreos/azure/README.md
----
kubectl-dash-f:
117: target file "examples/guestbook/redis-slave-controller.yaml" does not exist
118: target file "examples/guestbook/redis-slave-service.yaml" does not exist
FAIL: changes needed but not made due to --verify
/root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/docs/ is out of date. Please run hack/run-gendocs.sh
```

File: examples/guestbook/php-redis/index.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <link rel="stylesheet" href="//netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css">
     <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.2.12/angular.min.js"></script>
     <script src="/controllers.js"></script>
-    <script src="ui-bootstrap-tpls-0.10.0.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-ui-bootstrap/0.13.0/ui-bootstrap-tpls.js"></script>
```
Comment (brendandburns) [inline examples/guestbook/php-redis/index.html]: Maybe use the Google hosted mirror?

**Comment by brendandburns** [comment]
Basically LGTM one comment.  Please fix the tests.

**Comment by brendandburns** [comment]
Also needs rebase.

File: examples/guestbook/php-redis/index.html (hunk 1 above)
Comment (amygdala) [inline examples/guestbook/php-redis/index.html]: Working on tracking down the right link.

**Comment by amygdala** [comment]
Made some changes, but still working on fixing the tests.

**Comment by brendandburns** [comment]
Sorry, now needs a rebase...

@k8s-bot test this please

**Comment by amygdala** [comment]
I've rebased again.  I'm a bit suspicious of that Jenkins failure.  I don't *think* it's related to my changes.

**Comment by amygdala** [comment]
hah! the small intersection between "doesn't need rebasing" and "tests passing" has been achieved!

**Comment by amygdala** [comment]
Brendan, just a ping on this when you have time.  Amazingly it stiiiill doesn't need rebasing.

**Comment by amygdala** [comment]
Update: After a discussion w/ Brendan, am going to back out of the renaming. (Will keep the refactoring as is).  Stay tuned.

**Comment by amygdala** [comment]
Closing this-- see https://github.com/GoogleCloudPlatform/kubernetes/pull/12497 instead .

**Pull request closed without merging**